### PR TITLE
Fix support for empty attributes

### PR DESCRIPTION
**Pull request by tmelliottjr**
### Initial checklist

*   [x] I read the support docs 
*   [x] I read the contributing guide 
*   [x] I agree to follow the code of conduct 
*   [x] I searched issues and couldn’t find anything (or linked relevant results below) 
*   [x] If applicable, I’ve added docs and tests

### Description of changes

This change adds handling for attributes that are explicitly set with an empty string `""` or all whitespace `"  "`.

Prior to this, html comments that contained an empty string attribute were ignored completely. 

For example, given the following html `commentMarker` returns `null`.

```js
const html = '<!-- action target="1234" field="type" value="" -->'
const marker = commentMarker(node)
// marker === null
```

Additionally, attributes with whitespace only were coerced to `number` resulting in unexpected behavior.

For example, the following results in an attribute value of `0`.

```js
const html = '<!-- action target="1234" field="type" value=" " -->'
const marker = commentMarker(node)
// marker.parameters === {
//   target: 1234,
//   field: "type",
//   value: 0,
// }
```

### CODE REVIEW

**Review by wooorm — APPROVED** [review]

File: lib/index.js [hunk 1]
```diff
@@ -128,14 +128,15 @@ function parseParameters(value) {
   // eslint-disable-next-line max-params
   function replacer(_, $1, $2, $3, $4) {
     /** @type {MarkerParameterValue} */
-    let value = $2 || $3 || $4 || ''
+
+    let value = $2 ?? $3 ?? $4
```
Comment (wooorm) [inline lib/index.js:132]: Oh this one would be a breaking change, better to be safe. Not pretty but:

```suggestion
    let value = $2 === undefined ? ($3 === undefined ? $4 : $3) : $2;
```

**Comment by wooorm** [comment]
Thanks, released in [`2.1.2`](https://github.com/syntax-tree/mdast-comment-marker/releases/tag/2.1.2)!!

**Comment by mikesurowiec** [comment]
@wooorm @tmelliottjr thank you both for this! 👏

**Comment by tmelliottjr** [comment]
Thank you for the quick 👀 on this, greatly appreciated!

**Pull request merged**